config.c and history.c include both string.h and libgen.h. For this
reason, they have to conflict prototypes for basename(3). The workaround
here is to remove the include of libgen.h. The downside of this
workaround is that it causes dirname(3) to be implicitly declared.

--- a/src/config.c
+++ b/src/config.c
@@ -1,6 +1,5 @@
 #include <ctype.h>
 #include <errno.h>
-#include <libgen.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- a/src/history.c
+++ b/src/history.c
@@ -1,6 +1,5 @@
 #include <errno.h>
 #include <fcntl.h>
-#include <libgen.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
